services/src/datasets/storage.rs

```diff
@@ -133,7 +152,10 @@ pub enum VectorLoadingInfo {
 
 /// Handling of data sets provided by geo engine internally, staged and by external providers
 #[async_trait]
-pub trait DataSetDB: DataSetProvider + Send + Sync {
+pub trait DataSetDB: DataSetProvider + DataSetProviderDB + Send + Sync {}
+
+#[async_trait]
+pub trait DataSetProviderDB {
```

Comment:
Maybe you can add some info on public traits and types

Reply:
a0c4a7488f489f31a35aab8b045816f567874a98